browser/ui/sidebar/sidebar_controller.h

```diff
@@ -43,19 +43,30 @@ class SidebarController : public SidebarService::Observer,
   SidebarController(const SidebarController&) = delete;
   SidebarController& operator=(const SidebarController&) = delete;
 
+  // NOTE: Don't call this directly for panel item. Use ActivatePanelItem().
+  // This should be called as a result of SidePanelCoordinator's entry
+  // opening/closing event. If this method is called directly for activating
+  // panel, SidePanelCoordinator doesn't know about it.
+
   // |disposition| is only valid for shortcut type. If |disposition| is not
   // CURRENT_TAB, item at |index| is handled based on |disposition|.
   void ActivateItemAt(
       absl::optional<size_t> index,
       WindowOpenDisposition disposition = WindowOpenDisposition::CURRENT_TAB);
```

Comment:
It is my understanding that the only allowed caller for this is SidebarContainerView which is observing the entries at the moment. Then, how about renaming this to "OnEntryActivated", so that we can avoid confusion between ActivatePanelItem()? Or we might be able to use passkey pattern that allows only SidePanelEntryObservers to call this.

Reply:
This method is called for two purpose.
One is for [de]activating panel entry and it's only called from `SidebarContainerView` as a reaction for `OnEntryShow()` or `OnEntryHidden()` from `SidePanelCoordinator`.
The other is for loading shortcut type sidebar item. When this type of item is clicked item's url is loaded to current tab.

Not sure but I expect that upstream will make available observing SidePanelCoordinator(SidePanelUI) outside of view. Then SidebarController could be that observer.

I think this method could be splited into three methods.
`OnPanelEntryShown()` , `OnPanelEntryHidden()` and `LoadItemShortcutAt()`.
This could be refactored as a separated issue.